apis/python/src/tiledb/vector_search/module.cc

```diff
@@ -209,250 +269,419 @@ static void declare_nuv_query_heap_finite_ram(py::module& m, const std::string&
+template <
+    class T,
+    class shuffled_ids_type = uint64_t,
+    class indices_type = uint64_t>
 static void declare_dist_qv(py::module& m, const std::string& suffix) {
   m.def(("dist_qv_" + suffix).c_str(),
-      [](tiledb::Context& ctx,
-        const std::string& part_uri,
-        std::vector<int>& active_partitions,
-        ColMajorMatrix<float>& query,
-        std::vector<std::vector<int>>& active_queries,
-        std::vector<shuffled_ids_type>& indices,
-        const std::string& id_uri,
-        size_t k_nn,
-        uint64_t timestamp
-        /* size_t nthreads TODO: optional arg w/ fallback to C++ default arg */
+        [](tiledb::Context& ctx,
+           const std::string& part_uri,
+           std::vector<indices_type>& active_partitions,
+           ColMajorMatrix<T>& query,
+           // std::vector<std::vector<shuffled_ids_type>>& active_queries,
+           py::array& active_queries_arr,
+           std::vector<shuffled_ids_type>& indices,
+           const std::string& id_uri,
+           size_t k_nn,
+           uint64_t timestamp
+           /* size_t nthreads TODO: optional arg w/ fallback to C++ default arg */
         ) { /* TODO return type */
-            return detail::ivf::dist_qv_finite_ram_part<T, shuffled_ids_type>(
-                ctx,
-                part_uri,
-                active_partitions,
-                query,
-                active_queries,
-                indices,
-                id_uri,
-                k_nn,
-                timestamp);
-        }, py::keep_alive<1,2>());
+
+            size_t upper_bound{0};
+            auto nthreads = std::thread::hardware_concurrency();
+            auto temporal_policy{
+                (timestamp == 0) ?
+                    tiledb::TemporalPolicy() :
+                    tiledb::TemporalPolicy(tiledb::TimeTravel, timestamp)};
+
+            py::buffer_info buf_info = active_queries_arr.request();
+            auto shape = active_queries_arr.shape();
+            auto num_rows = shape[0];
+
+            auto active_queries = std::vector<std::vector<indices_type>>();
+            active_queries.reserve(num_rows);
+
+            auto ptr = static_cast<py::object*>(buf_info.ptr);
+
+            for (size_t i = 0; i < num_rows; ++i) {
+              py::list sublist = py::cast<py::list>(ptr[i]);
+              size_t sublist_length = py::len(sublist);
+              active_queries.emplace_back();
+              active_queries.back().reserve(sublist_length);
+              for (size_t j = 0; j < sublist_length; ++j) {
+                active_queries.back().emplace_back(py::cast<indices_type>(sublist[j]));
+              }
+            }
+
+  return detail::ivf::dist_qv_finite_ram_part<T, shuffled_ids_type>(
+      ctx,
+      part_uri,
+      active_partitions,
+      query,
+      active_queries,
+      indices,
+      id_uri,
+      k_nn,
+      timestamp);
+}, py::keep_alive<1,2>());
+#if 0
```

Comment:
Please avoid `#if 0` definitions. IMO they greatly reduce the readability of the code.

Reply:
Was keeping those for easy reference to previous implementation.  Will remove.